src/Extension/AssetTrait.php

```diff
@@ -180,16 +180,16 @@ private function normalizeAsset(FileAssetInterface $asset)
             return;
         }
 
-        $file = $this->getWebDirectory()->getFile($asset->getPath());
+        $app = $this->getContainer();
+
+        $file = $app['filesystem']->getFile("extensions_assets://{$this->getId()}/$path");
```

Comment:
It does make sense. It would be a bc break though. Also it is something each extension would have to define manually, not that that is a problem.